### ISSUE TITLE
IDE0059 is not raised inside a code block with using keyword.

### ISSUE THREAD

**Issue by GillesTourreau** [dotnet/roslyn#74219]
**Version Used**: SDK 8.0.302 ([4.11.0-2.final](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp.CodeStyle/4.11.0-2.final)

**Steps to Reproduce**:

Write the following code:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        using var _ = new MemoryStream();

        var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
    }

    public class Json
    {
    }
}
```

The `IDE0059` rule is not raised for the not used variable `jsonContent`.
But if we remove the `using` keyword:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        var _ = new MemoryStream();

        var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
    }

    public class Json
    {
    }
}
```

The `IDE0059` rule is raised to indicate that the `jsonContent` variable is not used.

We have the same issue if we use explicit code block for `using` keyword:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        using var _ = new MemoryStream();
        {
            var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
        }
    }

    public class Json
    {
    }
}
```

**Diagnostic Id**: [IDE0059](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0059): `Remove unnecessary value assignment`

**Expected Behavior**:

The `IDE0059` rule should be raised when a local variable is not used, even if there is a non-related `using` code block.
